src/prefect/deployments/steps/pull.py

```diff
@@ -93,6 +105,19 @@ def git_clone(
                 repository: git@github.com:org/repo.git
         ```
     """
+    if access_token and credentials:
+        raise ValueError(
+            "Please provide either an access token or credentials but not both."
+        )
+
+    if credentials:
+        if credentials.get("token"):
+            access_token = credentials.get("token")
+        elif credentials.get("username") and credentials.get("password"):
+            access_token = (
+                f"{credentials.get('username')}:{credentials.get('password')}"
+            )
```

Comment:
```suggestion
        if credentials.get("token"):
            access_token = credentials["token"]
        elif credentials.get("username") and credentials.get("password"):
            access_token = (
                f"{credentials['username']}:{credentials['password']}"
            )
```

Maybe a little more readable here since you already asked permission with the `if`?